### DISCUSSION TITLE
StreamingResponse does not work for Safari in live broadcasting?

### DISCUSSION THREAD

**Discussion by warkcod** [fastapi/fastapi#12009] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
async def tts(request: Request):

    from time import time as tt
    t1 = tt()
    if request.method == "GET":
        data = request.query_params
    else:
        data = await request.json()

    range_header = request.headers.get('Range')
    
    task:Base_TTS_Task = tts_synthesizer.params_parser(data)

    md5_value = task.md
    return_code = 200
    headers = {
            "content-type": 'audio/wav',
            "accept-ranges": "bytes",
            "Transfer-Encoding": "chunked",
    }
    if range_header:
        import re
        bytes_range = re.match(r'bytes=(\d+)-(\d+)?', range_header)
        if bytes_range:
            range_start = int(bytes_range.group(1))
            range_end = int(bytes_range.group(2)) if bytes_range.group(2) else None
            if range_end is not None:
                return_code = 206
                headers["content-length"] = str(range_end - range_start + 1)
                headers["content-range"] = f"bytes {range_start}-{range_end}/*"
               

    gen = tts_synthesizer.generate(task, return_type="numpy", range_header=range_header)
    return StreamingResponse(gen, status_code=return_code, headers=headers, media_type='audio/wav')
```

### Description

In Safari, we need to support Range, there is a discussion related to this:https://github.com/fastapi/fastapi/discussions/7718
but my situation is different, which in my case, the I use in live broadcasting in which content-length is unknown.
Safari only request the first 2 Bytes and stuck there. (While chrome works fine)
Is there any idea about this? 
Any help or clue is very appreciated, Thank you very much.

### Operating System

macOS, Other

### Operating System Details

MacOS, iOS

### FastAPI Version

0.104.1

### Pydantic Version

1.10.12

### Python Version

3.10.9

**Comment by dandiep** [comment]
Did you ever figure out a solution for this? I don't think it's a FastAPI problem, but surely there is some workaround...

**Comment by dandiep** [comment]
It seems you can stream MP3s to Safari but not WAVs. Here was my solution. Note: I just had PCM data, so this assumes there is no Wav header, so you would have to skip over that.

```
class PcmToMp3Iterator:
    def __init__(self, audio: Iterable[bytes]):
        self.audio = audio
        self.encoder = lameenc.Encoder()
        self.encoder.set_bit_rate(128)
        self.encoder.set_in_sample_rate(24000)
        self.encoder.set_channels(1)
        self.encoder.set_quality(2)
        self.flushed = False

    def __iter__(self):
        return self

    def __next__(self):
        if self.flushed:
            raise StopIteration()

        chunk_out = BytesIO()
        try:
            chunk = next(self.audio)

            if chunk is None:
                raise StopIteration()

            chunk_out.write(self.encoder.encode(chunk))

            chunk_out.seek(0)

            return chunk_out.read()
        except StopIteration:
            chunk_out.write(self.encoder.flush())
            self.flushed = True
            return chunk_out.read()
```

**Comment by YuriiMotov** [comment]
Let's close this for now to keep the discussions clean.
@warkcod, If you’ve found a solution, please share it here.
If you still need help, feel free to let us know.
